src/controller/ArchipelagoController.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  ArchipelagoController,
+  ArchipelagoControllerOptions,
+  ArchipelagoParameters,
+  Island,
+  Logger,
+  PeerData,
+  PeerPositionChange,
+  Position3D,
+  UpdateSubscriber,
+} from "../types/interfaces"
+
+import { fork, ChildProcess } from "child_process"
+import { GetIsland, WorkerMessage, WorkerRequest, WorkerResponse, WorkerStatus } from "../types/messageTypes"
+import { IdGenerator, sequentialIdGenerator } from "../misc/idGenerator"
+
+type SetPositionUpdate = { type: "set-position"; position: Position3D }
+type ClearUpdate = { type: "clear" }
+
+type PeerUpdate = SetPositionUpdate | ClearUpdate
+
+type WorkerControllerOptions = { requestTimeoutMs: number; workerLogging?: boolean }
+
+type PendingWorkerRequest = { resolve: (arg: any) => any; reject: (error: any) => any }
+
+export type WorkerOptions = { archipelagoParameters: ArchipelagoParameters; logging: boolean }
+class WorkerController {
```

Comment:
I see that everything on this file is public. Could we specify visibility for properties and methods?

Reply:
not EVERYTHING.

But yeah, maybe we should.